client.go

```diff
@@ -39,6 +37,29 @@ type Client struct {
 	// If not specified, a new [http.Client] with a Timeout of 1 minute will be used.
 	HTTP *http.Client
 
+	// APIKey allows specifying an APIKey to use for authentication.
+	// To use OAuth Client credentials, set ClientID and ClientSecret instead.
+	// To use identity federation, set ClientID and IDTokenFunc instead.
+	APIKey string
+	// ClientID is the ID of the Tailscale OAuth client.
+	// When set along with ClientSecret, the client will use OAuth client credentials for authentication.
+	// When set along with IDTokenFunc, the client will use identity federation for authentication.
+	// For OAuth, if empty and ClientSecret is provided, the ClientID will be derived from the ClientSecret.
+	// Overwrites APIKey if set.
+	ClientID string
```

Comment:
Instead of adding more and more parameters to the `Client` type, let's just make different pluggable authentication mechanisms and plug them into a single place like [here](https://github.com/tailscale/tailscale-client-go-v2/pull/60/files#diff-4b667feae66c9d46b21b9ecc19e8958cf4472d162ce0a47ac3e8386af8bbd8cfR43).